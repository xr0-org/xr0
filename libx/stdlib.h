#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
