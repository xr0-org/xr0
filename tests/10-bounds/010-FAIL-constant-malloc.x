#include <stdlib.h>

void
foo()
{
	int *p;
	p = malloc(5);
	p[0] = 5;
	p[1] = 4;
	p[2] = 3;
	p[3] = 2;
	p[4] = 1;
	p[5] = 0;
}
