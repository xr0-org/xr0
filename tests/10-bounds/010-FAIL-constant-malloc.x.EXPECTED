10-bounds/010-FAIL-constant-malloc.x:13:10: undefined indirection: out of bounds (foo)
